### PR TITLE
Enhanced unsubscribe/subscribe

### PR DESCRIPTION
**Pull request by phamann**
Quite a big PR i'm afraid, but some good things in there:

- Complete refactor the `Newsletter` component for enhanced experience on the client-side
- Add new `create` route to enable client-side subscription to newsletters
- Refactor clientside init'ing of react components within the alerts controller. I.e. use preact.

### CODE REVIEW

File: client/components/alerts/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { render } from 'preact';
+import { Section } from '@financial-times/n-section';
+import { Card } from '@financial-times/n-card';
+import MyftDigest from '../../../shared/components/alerts/myft-digest';
```
Comment (laurieboyes) [inline client/components/alerts/index.js]: Wearing my pedant hat 🎩  - could we fix the casing so it's `MyFtDigest`

File: server/controllers/alerts/create.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+import { subscribeUserToNewsletter } from '../../lib/email-web-services-client'
+
+module.exports = (req, res, next) => {
+	const userUuid = res.locals.userId;
+	const body = Object.assign({}, req.body, { userUuid });
+
+	subscribeUserToNewsletter(userUuid, body)
+		.then(() => {
+			res.status(201).json({message: `Sucessfullly subscribed to ${body.list}`});
```
Comment (laurieboyes) [inline server/controllers/alerts/create.js]: 'Sucessfullly' typo, should be 'Successfully'

Comment (keirog) [inline server/controllers/alerts/create.js]: Sucessfullly -> Successfullly

Comment (keirog) [inline server/controllers/alerts/create.js]: Thanks GitHub

File: server/controllers/alerts/delete.js [hunk 3]
```diff
@@ -8,6 +8,12 @@ module.exports = (req, res, next) => {
 	const listName = req.body.listName;
 
 	unsubscribeUserFromNewsletter(userId, listId)
-		.then(() => res.redirect(`/myft/confirm/${userId}/unsub-newsletter/?listName=${listName}`))
+		.then(() => {
+			if(req.method === 'POST') {
+				res.redirect(`/myft/confirm/${userId}/unsub-newsletter/?listName=${listName}`);
+			} else {
+				res.status(204).json({message: `Sucessfullly unsubscribed from ${listId}`});
```
Comment (keirog) [inline server/controllers/alerts/delete.js]: Sucessfullly -> Successfully

File: shared/components/alerts/newsletter/index.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+			'Registered': 'Free',
+			'Premium': 'Premium subscribers only',
+			'Standard': 'Subscribers only'
+		};
+
+		return subscriptions[item.subscriptionLevel];
+	}
+
+	getSubscriptionText(state) {
+		if(state.processing) { return 'Loading...'; }
```
Comment (keirog) [inline shared/components/alerts/newsletter/index.js]: `...` -> `…` ?

File: shared/components/alerts/newsletter/index.js [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+			'card--myft': true,
+			'card--myft-newsletter': true,
+			'card--empty': !item.imageUrl,
+			'card--disabled': !this.state.subscribed,
+			'js-myft-newsletter': true
+		});
+		const unsubscribeAction = this.getAction('unsubscribe').path;
+		const image = this.renderImage(item);
+		const content = this.renderContent(item);
+		const subscritionText = this.getSubscriptionText(this.state.subscribed);
```
Comment (laurieboyes) [inline shared/components/alerts/newsletter/index.js]: 🎩  subscritionText -> subscri**p**tionText

**Comment by laurieboyes** [comment]
How exciting 👍  (apart from one remaining comment)

File: shared/components/alerts/newsletter/index.js [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+						id={item.id}
+						isPressed={!this.state.subscribed}
+						formAction={unsubscribeAction}
+						hiddenInputs={[
+							{
+								name: 'listName',
+								value: item.name
+							}
+						]}
+						label={`${subscritionText} from ${item.name}`}
```
Comment (keirog) [inline shared/components/alerts/newsletter/index.js]: [Same](https://github.com/Financial-Times/next-myft-page/pull/883/files#r73168521)

**Comment by keirog** [comment]
👍  Yay

**Pull request merged**